### ISSUE TITLE
2.0.6 widgets error

### ISSUE THREAD

**Issue by henkiewie** [Lissy93/dashy#581]
<h3 dir="auto">Environment</h3>
<p dir="auto">Self-Hosted (Docker)</p>
<h3 dir="auto">Version</h3>
<p dir="auto">2.0.6</p>
<h3 dir="auto">Describe the problem</h3>
<p dir="auto">today at 12:07:47{<br>
today at 12:07:47  headers: {},<br>
today at 12:07:47  enableInsecure: false,<br>
today at 12:07:47  acceptCodes: 'null',<br>
today at 12:07:47  maxRedirects: 'null'<br>
today at 12:07:47}<br>
today at 12:07:47{<br>
today at 12:07:47  headers: {},<br>
today at 12:07:47  enableInsecure: false,<br>
today at 12:07:47  acceptCodes: 'null',<br>
today at 12:07:47  maxRedirects: 'null'<br>
today at 12:07:47}<br>
today at 12:07:47{<br>
today at 12:07:47  headers: {},<br>
today at 12:07:47  enableInsecure: true,<br>
today at 12:07:47  acceptCodes: 'null',<br>
today at 12:07:47  maxRedirects: 'null'<br>
today at 12:07:47}</p>
<h3 dir="auto">Additional info</h3>
<p dir="auto">updated to 2.0.6 yesterday (using latest label) gives errors in the log and the view.<br>
widgets are giving back values, but still give errors.<br>
turning off proxy makes no difefrence. (- type: gl-current-cpu useProxy: false).<br>
Going back to 2.0.5 solves the issue.<br>
<a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/74558874/160589748-f73939c3-8708-418a-beec-dc4a35d9e2b3.png"><img src="https://user-images.githubusercontent.com/74558874/160589748-f73939c3-8708-418a-beec-dc4a35d9e2b3.png" alt="Screenshot 2022-03-29 at 12 15 16" style="max-width: 100%;"></a></p>
<p dir="auto">console gives time-out;<br>
<g-emoji class="g-emoji" alias="warning" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/26a0.png">⚠️</g-emoji> Warning <g-emoji class="g-emoji" alias="warning" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/26a0.png">⚠️</g-emoji> Unable to fetch data This is likely not an issue with Dashy, but rather your configuration. If you think it is a bug, please open a ticket on GitHub: <a href="https://git.io/JukXk" rel="nofollow">https://git.io/JukXk</a><br>
CoolConsole.js:18 Stack Trace<br>
Error: timeout of 500ms exceeded<br>
i @ CoolConsole.js:18<br>
l @ ErrorHandler.js:27<br>
error @ WidgetMixin.js:75<br>
(anonymous) @ WidgetMixin.js:123<br>
Promise.catch (async)<br>
(anonymous) @ WidgetMixin.js:122<br>
makeRequest @ WidgetMixin.js:114<br>
fetchData @ GlancesMixin.js:26<br>
update @ WidgetMixin.js:67<br>
(anonymous) @ WidgetMixin.js:71<br>
2createError.js:16 Uncaught (in promise) Error: timeout of 500ms exceeded<br>
at t.exports (createError.js:16:15)<br>
at XMLHttpRequest.w.ontimeout (xhr.js:130:14)</p>
<h3 dir="auto">Please tick the boxes</h3>
<ul class="contains-task-list">
<li class="task-list-item"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked=""> You are using a <a href="https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions">supported</a> version of Dashy (check the first two digits of the version number)</li>
<li class="task-list-item"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked=""> You've checked that this <a href="https://github.com/Lissy93/dashy/issues?q=is%3Aissue">issue hasn't already been raised</a></li>
<li class="task-list-item"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked=""> You've checked the <a href="https://github.com/Lissy93/dashy/tree/master/docs#readme">docs</a> and <a href="https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting">troubleshooting</a> guide</li>
<li class="task-list-item"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked=""> You agree to the <a href="https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct">code of conduct</a></li>
</ul>

<blockquote>                    <img src="https://github.githubassets.com/images/modules/open_graph/github-logo.png" width="48" align="right" />                          <div>          <strong>                          <a href="https://github.com/">Build software better, together</a>                      </strong>        </div>                    <div>GitHub is where people build software. More than 73 million people use GitHub to discover, fork, and contribute to over 200 million projects.</div>                      </blockquote>

**Comment by Lissy93** [comment]
If you could also share the relevant parts of your config file, that would be helpful.

Looking at the error, looks like this part is the cause: `Error: timeout of 500ms exceeded`
You can try [setting timeout](https://github.com/Lissy93/dashy/blob/master/docs/widgets.md#setting-timeout) to a larger value, on the failing widgets, with: `timeout: 2000` 

<details>
	<summary>Example</summary>
	<p>

```yaml
- type: gl-current-cpu
  timeout: 5000
  options:
    hostname: https://glances.dns-device.local
```	

</p>
</details>

**Comment by Al4ndil** [comment]
Hi,
I have the same error but i don't use PI CPU. 
My config below
`appConfig:
  theme: material-dark
  fontAwesomeKey: 0821c65656
  enableFontAwesome: true
  layout: auto
  iconSize: medium
  language: fr
  auth:
    enableGuestAccess: false
    users: []
    enableKeycloak: false
  startingView: default
  defaultOpeningMethod: newtab
  statusCheck: true
  statusCheckInterval: 10
  faviconApi: allesedv
  routingMode: history
  enableMultiTasking: false
  webSearch:
    disableWebSearch: false
    searchEngine: google
    openingMethod: newtab
    searchBangs: {}
  hideComponents:
    hideHeading: false
    hideNav: false
    hideSearch: false
    hideSettings: false
    hideFooter: false
  showSplashScreen: false
  allowConfigEdit: true
  enableServiceWorker: false
  disableContextMenu: false
  disableUpdateChecks: false
  disableSmartSort: false
  enableErrorReporting: true
  widgetsAlwaysUseProxy: false
  preventWriteToDisk: false
  preventLocalSave: false
  disableConfiguration: false
pageInfo:
  title: Dashy
  description: Tableau de bord
  navLinks:
    - title: Home
      path: /
    - title: About
      path: /about
    - title: Source Code
      path: https://github.com/Lissy93/dashy
    - title: Wiki
      path: /
  footerText: ''
sections:
  - name: Administration
    items:
      - title: Dashboard SWAG
        icon: far fa-chart-network
        url: 
        id: 0_1478_dashboardswag
      - title: DNS1 AdGuard
        icon: hl-adguardhome
        url: http://10.0.0.253
        id: 2_1478_dnsadguard
      - title: DNS2 AdGuard
        icon: hl-adguardhome
        url: http://10.0.0.254
        id: 1_1478_dnsadguard
      - title: Livebox
        icon: hl-router
        url: http://192.168.1.1
        id: 3_1478_livebox
      - title: OMV
        description: ''
        icon: hl-openmediavault
        url: http://10.0.0.101:90/
        id: 4_1478_omv
      - title: Portainer
        description: ''
        icon: hl-portainer
        url: http://10.0.0.102:9000
        id: 5_1478_portainer
      - title: Proxmox
        description: ''
        icon: hl-proxmox
        url: https://10.0.0.100:8006/
        id: 6_1478_proxmox
      - title: Routeur Netgear
        icon: hl-netgear
        url: http://10.0.0.1
        id: 7_1478_routeurnetgear
      - title: Seedbox
        icon: hl-rutorrent
        url: 
        id: 8_1478_seedbox
    icon: fad fa-toolbox
    displayData:
      sortBy: alphabetical
      rows: 1
      cols: 1
      collapsed: false
      hideForGuests: false
  - name: Cloud
    displayData:
      sortBy: alphabetical
      rows: 1
      cols: 1
      collapsed: false
      hideForGuests: false
    items:
      - title: Duplicati
        icon: hl-duplicati
        url: http://10.0.0.104:8200/
        id: 0_503_duplicati
      - title: Nextcloud
        icon: hl-nextcloud
        url: 
        id: 1_503_nextcloud
      - title: Paperless-ng
        id: 2_503_paperlessng
    icon: ''
  - name: MultimÃ©dia
    displayData:
      sortBy: alphabetical
      rows: 1
      cols: 1
      collapsed: false
      hideForGuests: false
    items:
      - title: Jackett
        icon: hl-jackett
        url: http://10.0.0.101:9117
        id: 0_1167_jackett
      - title: Jellyfin
        icon: hl-jellyfin
        url: http://10.0.0.101:8096
        id: 1_1167_jellyfin
      - title: Overseerr
        icon: hl-overseerr
        url: http://10.0.0.101:5055
        id: 2_1167_overseerr
      - title: Plex
        icon: hl-plex
        url: http://10.0.0.101:32400
        id: 3_1167_plex
      - title: Prowlarr
        icon: hl-prowlarr
        url: http://10.0.0.101:9696
        id: 4_1167_prowlarr
      - title: Radarr
        icon: hl-radarr
        url: http://10.0.0.101:7878
        id: 5_1167_radarr
      - title: Sonarr
        description: Gestion des mangas
        icon: hl-sonarr
        url: http://10.0.0.101:9090
        statusCheck: false
        id: 7_1167_sonarr
      - title: Sonarr
        description: Gestion des sÃ©ries
        icon: hl-sonarr
        url: http://10.0.0.101:8989
        statusCheck: false
        id: 6_1167_sonarr
    icon: fad fa-film
`

**Comment by Lissy93** [comment]
@Al4ndil, Just up the timeout

**Comment by Al4ndil** [comment]
> @Al4ndil, Just up the timeout

Sorry but can you explain me more how to make this in my case ?! I don't use widget and i don't know where i must put "timeout : 2000"...

**Comment by Lissy93** [comment]
Sure, there's an example above in [this comment](https://github.com/Lissy93/dashy/issues/581#issuecomment-1081937748), I'll paste it below for ya, but basically just put `timeout: 5000` in whichever widget is failing.

This is explained more here, in the **[widget docs](https://github.com/Lissy93/dashy/blob/master/docs/widgets.md#setting-timeout)**.
If you're ever unsure of where something goes, you can reference the **[configuring docs](https://github.com/Lissy93/dashy/blob/master/docs/configuring.md)**.

<div type='discussions-op-text'>

<sup>Originally posted by **lissy93** March 29, 2022</sup>

Looking at the error, looks like this part is the cause: `Error: timeout of 500ms exceeded`
You can try [setting timeout](https://github.com/Lissy93/dashy/blob/master/docs/widgets.md#setting-timeout) to a larger value, on the failing widgets, with: `timeout: 2000` 

<details>
	<summary>Example</summary>
	<p>

```yaml
- type: gl-current-cpu
  timeout: 5000
  options:
    hostname: https://glances.dns-device.local
```	

</p>
</details>

</div>

**Comment by Al4ndil** [comment]
@Lissy93 I'm sorry again but **i don't use any widget**. So where i need to put timeout ?
Perhaps i don't understand but you mentionned "widget" each time you mentionned "timeout".

**Comment by Lissy93** [comment]
This ticket is about widget timeout errors, so not sure how you are getting this error message if you don't have a widget...?

Sorry, don't think I understand what you mean, could you explain the issue further? It might be helpful to see a screenshot, and also any console messages (here's [instructions](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#how-to-open-browser-console) if you're unsure how)

**Comment by Al4ndil** [comment]
In this pastebin, there is all my conf.yml 
https://pastebin.com/2uNf14aG
And i had the same error with 2.0.6
https://github.com/Lissy93/dashy/issues/581#issue-1184638692
I resolve this with a downgrade to 2.0.5

**Comment by Lissy93** [comment]
Could you share:
- A screenshot of the issue
- The output of the console
  - (it can't be the same as above, as that's a widget error, and you don't have any widgets).

You config looks fine 👍

**Comment by Al4ndil** [comment]
No problem on the web ui.
[https://i.imgur.com/mECLGFk.png](url)
Brave's console
> o937511.ingest.sentry.io/api/5887934/envelope/?sentry_key=3138ea85f15a4fa883a5b27a4dc8ee28&sentry_version=7:1          Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ 
Dashy  V2.0.6 🚀
o937511.ingest.sentry.io/api/5887934/envelope/?sentry_key=3138ea85f15a4fa883a5b27a4dc8ee28&sentry_version=7:1          Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
o937511.ingest.sentry.io/api/5887934/envelope/?sentry_key=3138ea85f15a4fa883a5b27a4dc8ee28&sentry_version=7:1          Failed to load resource: net::ERR_BLOCKED_BY_CLIENT

**Comment by Lissy93** [comment]
Doesn't seem to be any errors.
What problem are you experiencing if it's not on the UI?

**Comment by Lissy93** [comment]
Sorry but I'm very confused....
This ticket is about widget timeout errors, but you said you're not using any widgets. And you don't have any issues on the UI, nor any errors in the console. So it sounds like everything's working fine. Without a description of the issue, I cannot help.

**Comment by Al4ndil** [comment]
Hmmm the errors are on the logs of Dashy
[https://i.imgur.com/z4D8VuC.png](url)
I don't have these errors with 2.0.5

**Comment by Lissy93** [comment]
Ahhhh, lol now know hat you mean. I thought u were trolling me!

So those aren't errors, they're nothing to worry about, it's described in more detail in the [Troubleshooting Guide](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#warnings-in-the-console-during-deploy), but is just info.

**Comment by Al4ndil** [comment]
No no no ! I wouldn't have allowed myself to troll you :)

Thx for your answer ! :)

**Comment by henkiewie** [comment]
Hi Lissy,

There seems to be an error in the 2.0.7 version using the the item target openingmethod. It  is not used anymore and is falling back to the default in `appconfig.defaultOpeningMethod: newtab`. So if the item is target: modal and the default is newtab its using newtab and vice versa. Going back to 2.0.6 solves it for me.
I see no errors in the browser console. (Some of my docker containers use their own web server dissallowing Xframe options so modal is not an option). Want me to make a bug report? I think the bot is closing it automatic 😄 .

Greetings Henk.

**Comment by Lissy93** [comment]
Oops, my bad, I did a LOT of refactoring for items, and looks like I messed up the target attribute, thanks for raising.
I pushed a fix out in a784a2fe308508ee23062c8a974dd679a16cd0a0 which is now merged :)

**Comment by henkiewie** [comment]
Problem fixed. Have a nice (easter) weekend.

Greetings,
Henk
